pytermgui/ansi_interface.py

```diff
@@ -107,243 +142,263 @@ class Color:
         self._listeners: dict[int, list[Callable[..., Any]]] = {}
-
         # TODO: Windows doesn't support SIGWINCH, is there another alternative?
-        if hasattr(signal, "SIGWINCH"):
+        if _IS_NT:
+            asyncio.run(self._WinSIGWINCH())
+        else:
             signal.signal(signal.SIGWINCH, self._update_size)
 
-    def _call_listener(self, event: int, data: Any) -> None:
-        """Call event callback is one is found."""
+    async def _WinSIGWINCH(self) -> None:
+        while True:
+            n_width = get_terminal_size()
+            if n_width != self.size:
+                self._update_size(28, (sys._getframe(1) if _SYS_HAS_FRAME else None))
+            await asyncio.sleep(0.5)
 
+    def _call_listener(self, event: int, data: Any) -> None:
+        """
+        Call event callback is one is found.
+        """
         if event in self._listeners:
             for callback in self._listeners[event]:
                 callback(data)
 
-    def _get_size(self) -> tuple[int, int]:
-        """Get screen size while substracting the origin position"""
-
-        # This always has len() == 2, but mypy can't see that.
-        return tuple(val - org for val, org in zip(screen_size(), self.origin))  # type: ignore
-
-    def _update_size(self, *_: Any) -> None:
-        """Resize terminal when SIGWINCH occurs.
-
-        Note:
-            SIGWINCH is not supported on Windows, so this isn't called."""
-
-        self.size = self._get_size()
+    def _update_size(self, signum: int, frame: FrameType) -> None:
+        """
+        Resize terminal when SIGWINCH or _WinSIGWINCH occurs
+        """
+        self.size = get_terminal_size()
         self._call_listener(self.RESIZE, self.size)
 
     @property
     def width(self) -> int:
         """Get width of terminal"""
-
         return self.size[0]
 
     @property
     def height(self) -> int:
         """Get height of terminal"""
-
         return self.size[1]
 
     def subscribe(self, event: int, callback: Callable[..., Any]) -> None:
-        """Subscribe a callback function to an event
+        """
+        Subscribe a callback function to an event
 
         The callback takes an event-specific argument payload:
-            RESIZE: tuple[int, int] - New screen size"""
-
+            RESIZE: tuple[int, int] - New screen size
+        """
         if not event in self._listeners:
             self._listeners[event] = []
-
         self._listeners[event].append(callback)
 
     def fill(self, color: int = 0, flush: bool = True) -> None:
-        """Fill entire terminal with a color"""
-
+        """
+        Fill entire terminal with a color
+        """
+        _fill = background(" " * (self.width - 1), color)
         for height in range(self.height):
-            sys.stdout.write(
-                f"\033[{height};0H" + background(" " * (self.width - 1), color)
-            )
-
+            sys.stdout.write(f"\x1b[{height};0H{fill}")
```

Comment:
Interesting change, I would use `buffer` or something along those lines instead of `fill` for the name, though.

Also not really sure if this is any better in terms of speed.

Reply:
I used `_fill` without thinking 😅. I was in hurry so I didn't checked properly and added random variable name in speed